markdown_it/tree.py

```diff
@@ -44,6 +45,12 @@ def __init__(self) -> None:
         # children (i.e. inline or img)
         self.children: List["SyntaxTreeNode"] = []
 
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self.type})"
+
+    def __getitem__(self, item: int) -> "SyntaxTreeNode":
```

Comment:
I personally favor having just one way to do things, and would prefer the more explicit `.children[i]` syntax. Don't feel too strongly about that though.

If we keep this though, we might want to improve type annotations. `item` can be a `slice` in addition to `int` and return type will change accordingly too. An `@overload` is probably needed.

The return type will also break if subclassed. Merging https://github.com/executablebooks/markdown-it-py/pull/131 and changing the type to `_T` (and annotating `self` as `_T`) should fix this.

Reply:
yeh exactly, I was going to leave this type annotation stuff to the other PR